### ISSUE TITLE
(cstor-target): maya-volume-exporter container experiencing  `panic: runtime error: invalid memory address or nil pointer dereference`

### ISSUE THREAD

**Issue by dargasudarshan** [openebs/openebs#2341]
## Description

- maya-volume-exporter container experiencing panic `panic: runtime error: invalid memory address or nil pointer dereference`

## Expected Behavior

- Container should run seamlessly without any panics.

## Current Behavior

- maya-volume-exporter container experiencing panic `panic: runtime error: invalid memory address or nil pointer dereference`

## Steps to Reproduce

- Deploy Nuodb application on cstor-sparse-pool.
- Run ycsb load on top of the volume.
>> Here observed maya-volume-exporter container experiencing panic `panic: runtime error: invalid memory address or nil pointer dereference`

## Your Environment

* `kubectl get nodes`:
```
kubectl get nodes
NAME                                             STATUS    ROLES     AGE       VERSION
gke-e2e-cstor-dev-1-default-pool-224a4771-9nxq   Ready     <none>    2h        v1.11.5-gke.5
gke-e2e-cstor-dev-1-default-pool-224a4771-pqg7   Ready     <none>    2h        v1.11.5-gke.5
gke-e2e-cstor-dev-1-default-pool-224a4771-txfx   Ready     <none>    2h        v1.11.5-gke.5
```
* `kubectl get pods --all-namespaces`:
```
kubectl get pods --all-namespaces
NAMESPACE     NAME                                                              READY     STATUS      RESTARTS   AGE
kube-system   heapster-v1.6.0-beta.1-78cb745f97-qkks9                           2/2       Running     0          18h
kube-system   kube-dns-7549f99fcc-2x7zf                                         4/4       Running     0          18h
kube-system   kube-dns-7549f99fcc-wg5vs                                         4/4       Running     0          18h
kube-system   kube-dns-autoscaler-67c97c87fb-sjphq                              1/1       Running     0          18h
kube-system   kube-proxy-gke-e2e-cstor-dev-1-default-pool-224a4771-9nxq         1/1       Running     0          2h
kube-system   kube-proxy-gke-e2e-cstor-dev-1-default-pool-224a4771-pqg7         1/1       Running     0          2h
kube-system   kube-proxy-gke-e2e-cstor-dev-1-default-pool-224a4771-txfx         1/1       Running     0          2h
kube-system   l7-default-backend-5bc54cfb57-5vqj2                               1/1       Running     0          18h
kube-system   metrics-server-v0.2.1-fd596d746-slhw4                             2/2       Running     0          18h
litmus        litmus-nuodb-28nn6-w9hcs                                          0/1       Completed   0          1h
maya-system   cortex-agent-6b5b4f87f7-gj6x7                                     1/1       Running     0          18h
maya-system   fluentd-aggregator-7f4776ccbc-8dznh                               1/1       Running     0          18h
maya-system   fluentd-forwarder-f6t2x                                           1/1       Running     3          2h
maya-system   fluentd-forwarder-gfxh7                                           1/1       Running     3          2h
maya-system   fluentd-forwarder-hpqpp                                           1/1       Running     2          2h
maya-system   maya-io-agent-7gj6g                                               1/1       Running     0          2h
maya-system   maya-io-agent-l6bpt                                               1/1       Running     0          2h
maya-system   maya-io-agent-vk2zr                                               1/1       Running     0          2h
maya-system   openebs-monitor-plugin-6bcdb95fb4-c99kg                           2/2       Running     0          18h
maya-system   upgrade-agent-1547016300-j6m4c                                    0/1       Completed   0          8m
maya-system   weave-scope-agent-jgm5l                                           1/1       Running     0          2h
maya-system   weave-scope-agent-lm596                                           1/1       Running     0          2h
maya-system   weave-scope-agent-t85hh                                           1/1       Running     0          2h
maya-system   weave-scope-app-7c7c597847-qc6mx                                  1/1       Running     0          18h
nuodbns       admin-0                                                           1/1       Running     0          1h
nuodbns       nuodb-insights                                                    2/2       Running     0          1h
nuodbns       sm-0                                                              1/1       Running     0          1h
nuodbns       te-8487454c7c-9ct5c                                               1/1       Running     0          1h
nuodbns       ycsb-load-vbddv                                                   1/1       Running     0          1h
openebs       cstor-sparse-pool-0vnf-5578f86545-fgzxt                           2/2       Running     0          2h
openebs       cstor-sparse-pool-hg1o-7dc6498cfb-8ngh7                           2/2       Running     0          2h
openebs       cstor-sparse-pool-z8sa-69546776b7-6wf7w                           2/2       Running     0          2h
openebs       maya-apiserver-6bcc5d9b5f-2ttb7                                   1/1       Running     0          2h
openebs       openebs-ndm-nfvw7                                                 1/1       Running     0          2h
openebs       openebs-ndm-s62rb                                                 1/1       Running     0          2h
openebs       openebs-ndm-xmksn                                                 1/1       Running     0          2h
openebs       openebs-provisioner-5c65ff5d55-k274j                              1/1       Running     0          2h
openebs       openebs-snapshot-operator-9898bbb95-r7hff                         2/2       Running     0          2h
openebs       pvc-fe48fc4b-13d0-11e9-9fa1-42010a800134-target-788db8cfc5wt8rh   3/3       Running     10         1h
openebs       pvc-ffbc38f5-13d0-11e9-9fa1-42010a800134-target-65b9c6d966plcpt   3/3       Running     10         1h
```
* `kubectl get services`:
* `kubectl get sc`:
* `kubectl get pv`:
* `kubectl get pvc`:
* OS (from `/etc/os-release`):
```
cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
* Kernel (from `uname -a`):
* Install tools:
* Others:
Panic log:
```
I0109 06:35:40.565823       1 cstorcollector.go:156] Got response: "IOSTATS  { \"iqn\": \"iqn.2016-09.com.openebs.cstor:pvc-fe48fc4b-13d0-11e9-9fa1-42010a800134\", \"WriteIOPS\": \"214\", \"ReadIOPS\": \"264\", \"TotalWriteBytes\": \"154042368\", \"TotalReadBytes\": \"6550528\", \"Size\": \"5368709120\", \"UsedLogicalBlocks\": \"183\", \"SectorSize\": \"512\", \"UpTime\": \"3341\", \"TotalReadTime\": \"7771443068\", \"LUTotalReadTime\": \"6526907663\", \"ReplicationTotalReadTime\": \"6524104817\", \"TotalWriteTime\": \"104963045946\", \"LUTotalWriteTime\": \"67935235592\", \"ReplicationTotalWriteTime\": \"66029634769\", \"TotalReadBlockCount\": \"12794\", \"TotalWriteBlockCount\": \"300864\", \"ReplicaCounter\": 3, \"RevisionCounter\": \"534\", \"Status\": \"Healthy\", \"Replicas\": [ { \"replicaId\": \"2528929910583480344\", \"Address\": \"10.20.33.5\", \"Mode\": \"Healthy\", \"checkpointedIOSeq\": \"0\", \"inflightRead\": \"0\", \"inflightWrite\": \"0\", \"inflightSync\": \"0\", \"upTime\": 3327, \"ReadReqTime\": \"1396023\", \"ReadRespTime\": \"376524688\", \"WriteReqTime\": \"26837625860\", \"WriteRespTime\": \"36156169822\" }, { \"replicaId\": \"14171238440616855008\", \"Address\": \"10.20.32.17\", \"Mode\": \"Healthy\", \"checkpointedIOSeq\": \"0\", \"inflightRead\": \"0\", \"inflightWrite\": \"0\", \"inflightSync\": \"0\", \"upTime\": 3325, \"ReadReqTime\": \"4101172\", \"ReadRespTime\": \"337041312\", \"WriteReqTime\": \"10589336686\", \"WriteRespTime\": \"37055800921\" }, { \"replicaId\": \"14264725128611374967\", \"Address\": \"10.20.34.4\", \"Mode\": \"Healthy\", \"checkpointedIOSeq\": \"0\", \"inflightRead\": \"0\", \"inflightWrite\": \"0\", \"inflightSync\": \"0\", \"upTime\": 3324, \"ReadReqTime\": \"4354828651\", \"ReadRespTime\": \"5040391545\", \"WriteReqTime\": \"11534964482\", \"WriteRespTime\": \"22337208449\" } ] }\r\nOK IOSTATS\r\n"
I0109 06:35:41.562784       1 cstorcollector.go:156] Got response: "de\": \"Healthy\", \"checkpointedIOSeq\": \"0\", \"inflightRead\": \"0\", \"inflightWrite\": \"0\", \"inflightSync\": \"0\", \"upTime\": 3326, \"ReadReqTime\": \"4101172\", \"ReadRespTime\": \"337041312\", \"WriteReqTime\": \"10589336686\", \"WriteRespTime\": \"37055800921\" }, { \"replicaId\": \"14264725128611374967\", \"Address\": \"10.20.34.4\", \"Mode\": \"Healthy\", \"checkpointedIOSeq\": \"0\", \"inflightRead\": \"0\", \"inflightWrite\": \"0\", \"inflightSync\": \"0\", \"upTime\": 3325, \"ReadReqTime\": \"4354828651\", \"ReadRespTime\": \"5040391545\", \"WriteReqTime\": \"11534964482\", \"WriteRespTime\": \"22337208449\" }
] }\r\nOK IOSTATS\r\n"
E0109 06:35:41.562849       1 cstorcollector.go:125] Error in unmarshalling, found error: invalid character 'd' looking for beginning of value
E0109 06:35:41.562879       1 cstorcollector.go:55] Error in connection, closing the connection
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x78e24b]

goroutine 1759 [running]:
github.com/openebs/maya/cmd/maya-exporter/app/collector.(*Cstor).reader(0xc0000c3450, 0x0, 0x0, 0xc00010ccc0, 0xc00008a700)
        /home/travis/gopath/src/github.com/openebs/maya/cmd/maya-exporter/app/collector/cstorcollector.go:88 +0x9b
github.com/openebs/maya/cmd/maya-exporter/app/collector.(*Cstor).set(0xc0000c3450, 0xc0000c3470, 0xc0002bf680, 0x4)
        /home/travis/gopath/src/github.com/openebs/maya/cmd/maya-exporter/app/collector/cstorcollector.go:152 +0xd4
github.com/openebs/maya/cmd/maya-exporter/app/collector.(*Cstor).collector(0xc0000c3450, 0xc0000c3470, 0xc000041e98, 0x42c3e1)
        /home/travis/gopath/src/github.com/openebs/maya/cmd/maya-exporter/app/collector/cstorcollector.go:53 +0x57
github.com/openebs/maya/cmd/maya-exporter/app/collector.(*VolumeStatsExporter).Collect(0xc0000c3440, 0xc00010d020)
        /home/travis/gopath/src/github.com/openebs/maya/cmd/maya-exporter/app/collector/metrics.go:398 +0x31f
github.com/openebs/maya/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc000028f80, 0xc00010d020, 0x933840, 0xc0000c3440)
        /home/travis/gopath/src/github.com/openebs/maya/vendor/github.com/prometheus/client_golang/prometheus/registry.go:433 +0x61
created by github.com/openebs/maya/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
        /home/travis/gopath/src/github.com/openebs/maya/vendor/github.com/prometheus/client_golang/prometheus/registry.go:431 +0x302
```
OpenEBS version: CI

**Comment by dargasudarshan** [comment]
Attaching logs:
[maya-volume-exporter.log](https://github.com/openebs/openebs/files/2739703/maya-volume-exporter.log)

**Comment by utkarshmani1997** [comment]
This issue has been fixed in openebs/maya#873
